snippet/snippet.go

```diff
@@ -43,32 +41,41 @@ func (snippets *Snippets) Load(includeDirs bool) error {
 				`snippet file not found. %s
 Please run 'pet configure' and provide a correct file path, or remove this
 if you only want to provide snippetdirs instead`,
-				snippetFile,
+				config.Conf.General.SnippetFile,
 			)
 		}
 	}
 
-	// Optionally load snippets from snippet directories
-	if includeDirs {
-		for _, dir := range config.Conf.General.SnippetDirs {
-			if _, err := os.Stat(dir); err != nil {
-				if os.IsNotExist(err) {
-					return fmt.Errorf("snippet directory not found. %s", dir)
-				}
-				return fmt.Errorf("failed to load snippet directory. %v", err)
+	for _, snippetDir := range config.Conf.General.SnippetDirs {
```

Comment:
I think you removed the includeDirs if statement by mistake. That's actually very important - we don't always want to load all the snippets.

Specifically, when creating a new snippet, we want to load the snippets from the main file only, count the lines, and then add the new snippet to that subset of snippets (main file snippets only) and then save them all into the main file.

If we didn't have includeDirs we would actually save the snippets from ALL dirs into the main file 😓  uncovered this bug recently.

Reply:
You're right. I definitely did that. Easy enough a fix